src/vs/platform/telemetry/common/telemetryUtils.ts

```diff
@@ -286,6 +286,12 @@ export function getPiiPathsFromEnvironment(paths: IPathEnvironment): string[] {
  * @returns The cleaned stack
  */
 function anonymizeFilePaths(stack: string, cleanupPatterns: RegExp[]): string {
+	// Detect if it's a user agent and don't sanitize it
+	// As the / is a false positive for file paths
+	if (stack.indexOf('Mozilla/') >= 0) {
```

Comment:
is this actually a safe check. Shouldn't we be more robust in ensuring this isn't a file path. I could very well work at Mozilla and have a path with that included.

Reply:
Makes sense. I updated the check to see if it starts with since a path on windows would be `DriveLetter:\` and on linux would be `/` there is still the unlikely chance that you have some path without a preceding slash `Mozilla/Documents` or something, but I'm not sure we've ever seen that. I looked at https://developers.whatismybrowser.com/ to determine that the major browsers we care about their user agents are either in the form `Mozilla/` or `Opera/` then with additional stuff after. If we want to get more specific I can look into a better regex for this. The reason this bug arose is we apply path cleaning now to all properties not just errors so we're actually being even more strict than before

Reply:
How about checking for the actual user agent string (`navigator.userAgent`) here?